plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      s.SchedulerName,
```

Comment:
e.g. make LockObjectNamespace and LockObjectName fields in options.SchedulerServer ~= previous comment - https://github.com/kubernetes/kubernetes/pull/42961#issuecomment-286753197 so yeah +1 ;-)